src/pages/index.js

```diff
@@ -0,0 +1,372 @@
+
+  cardImageEl.addEventListener("click", () => {
+    openModal(previewModal);
+    previewModalCaptionEl.textContent = data.name;
+    previewModalImageEl.src = data.link;
+    previewModalImageEl.alt = data.name;
+  });
+
+  cardDeleteBtn.addEventListener("click", () => 
+    handleDeleteCard(cardElement, data._id, cardDeleteBtn));
+
+  return cardElement;
+}
+
+function handleDeleteCard(cardElement, cardId, cardDeleteBtn) {
+  selectedCard = cardElement;
+  selectedCardId = cardId;
+  openModal(deleteModal);
+}
+
+function handleDeleteSubmit(evt) {
+  evt.preventDefault();
+
+  setButtonText(avatarDeletetBtn, true, "Delete", "Deleting...");
+
+  api.deleteCard(selectedCardId)
+    .then(() => {
+      selectedCard.remove();
+      closeModal(deleteModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(avatarDeletetBtn, false, "Delete");
+    });
+}
+
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+  const profileSubmitBtn = evt.submitter;
+
+  setButtonText(profileSubmitBtn, true, "Save", "Saving...");
+
+  const updatedUserInfo = { 
+    name: editModalNameInput.value,
+    about: editModalAboutInput.value 
+  };
+
+  api.editUserInfo(updatedUserInfo)  
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      editFormElement.reset();
+      closeModal(editModal);
+
+      api.getAppInfo()
+        .then(([cards, userInfo]) => {
+          profileName.textContent = userInfo.name;
+          profileDescription.textContent = userInfo.about;
+        })
+        .catch(console.error);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(profileSubmitBtn, false, "Save");
+    });
+}
+
+function toggleSubmitButton() {
+  const isInputValid = cardNameInput.value.trim() !== "" && cardLinkInput.value.trim() !== "";
+  cardSubmitBtn.disabled = !isInputValid; 
+}
+
+cardNameInput.addEventListener("input", toggleSubmitButton);
+cardLinkInput.addEventListener("input", toggleSubmitButton);
+
+function handleAddCardSubmit(evt) {
+  evt.preventDefault();
+  cardSubmitBtn.disabled = true; 
+  setButtonText(cardSubmitBtn, true, "Save", "Saving...");
+
+  const inputValues = { name: cardNameInput.value, link: cardLinkInput.value };
+
+  api.createCard(inputValues)
+    .then((data) => {
+      const cardElement = getCardElement(data);
+      cardsList.prepend(cardElement);
+      cardForm.reset(); 
+      closeModal(cardModal); 
+      cardSubmitBtn.disabled = true; 
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(cardSubmitBtn, false, "Save"); 
+    });
+}
+
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+  setButtonText(avatarSubmitBtn, true, "Save", "Saving...");
+
+  api.editAvatarInfo(avatarInput.value)  
+    .then((data) => {
+      avatarImage.src = data.avatar;
+      avatarForm.reset();
+      closeModal(avatarModal);
+
+      api.getAppInfo()
```

Comment:
* Please, delete ` api.getAppInfo()` from the `avatar`  submission because you get everything in the response from the server. You don't need to fetch everything again


```js

function handleAvatarSubmit(evt) {
  evt.preventDefault();
  setButtonText(avatarSubmitBtn, true, "Save", "Saving...");

  api
    .editAvatarInfo(avatarInput.value)
    .then((data) => {
      avatarImage.src = data.avatar;
      avatarForm.reset();
      disabledButton(avatarSubmitBtn, settings);
      closeModal(avatarModal);
    })
    .catch(console.error)
    .finally(() => {
      setButtonText(avatarSubmitBtn, false, "Save");
    });
}
```